src/java.desktop/share/classes/sun/print/RasterPrinterJob.java

```diff
@@ -2393,6 +2409,9 @@ protected int printPage(Pageable document, int pageIndex)
                         painterGraphics.setDelegate((Graphics2D) bandGraphics.create());
                         painter.print(painterGraphics, origPage, pageIndex);
                         painterGraphics.dispose();
+                        if (monochrome) {
+                            monochromeConverter.filter(band, band);
```

Comment:
Like I mentioned elsewhere, I think a greyscale image would mean you don't need to do this.
However the code that sends the image to the printer  needs a version that handles that format.
Currently the Postscript code handles only BGR .. 

Also, this code is used on WINDOWS as well as other platforms, and monochrome is handled natively by the printer, so this looks like it is doing things that are unnecessary on that platform.